projects/plugins/jetpack/modules/widgets/contact-info.php

```diff
@@ -163,7 +163,10 @@ public function widget( $args, $instance ) {
 				}
 			}
 
-			if ( is_email( trim( $instance['email'] ) ) ) {
+			if (
+				! empty( $instance['email'] )
```

Comment:
It looks like `$instance['email']` should always exist thanks to line 108, so the `empty()` is partially redundant. Unnecessary use of `empty()` is a pet peeve of mine. 🙂

Just this should be equivalent
```suggestion
				$instance['email']
```
although this may be more robust
```suggestion
				is_string( $instance['email'] )
```

Reply:
1. `! empty( $instance['email'] )`: prevents `is_email()` call for both `null` and `''`
2. `$instance['email']`: calls `is_email()` on any truthy value, and may fail if it's an unexpected non-string type
3. `is_string( $instance['email'] )`: calls `is_email()` on any string value
4. `is_string( $instance['email'] ) && $instance['email'] !== ''`: same practical effect as 1, but less concise

I suspect the `empty()` call is slightly more performant than the `is_email()` call that uses regex (though probably has some short-circuit for empty content). Is 3 still preferable to you?

Reply:
Since `! empty( $instance['email'] )` is equivalent to `isset( $instance['email'] ) && $instance['email']`, and `isset( $instance['email'] )` will always be true here, 1 and 2 do the same thing.

I'm ok with any of 2–4, since getting any non-string other than null seems unlikely, and `''` and `'0'` aren't valid email addresses. Really, if you insist on 1, I won't block it because it's not worth arguing over.

I suspect worrying about the performance of the options here is a microoptimization, is this really a hot path? Also I note that `is_email()` does indeed [have an early check for empty content](https://github.com/WordPress/wordpress-develop/blob/626d9af280566d008ccf501952f119fb0ce61662/src/wp-includes/formatting.php#L3538-L3554) anyway.

Reply:
I've no strong opinion here; I've gone ahead with 2. :^)

> I suspect worrying about the performance of the options here is a microoptimization

I agree; I was mainly curious about your pet peeve against `empty()`. Using `empty()` is a safeguard against future errors as well (e.g. if code changed and the `isset()` was no longer always `true`). The `is_email()` call always calls `empty()` as well. :^D

Reply:
OTOH, being in the habit of using `empty()` unnecessarily means you wouldn't get any runtime warning if you typoed something like `$instanc['email']` or `$instance['emial']`.